src/components/Home.js

```diff
@@ -1,10 +1,58 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { countryDetails, getCountriesData } from '../redux/countrySlice';
+import './Home.css';
+import Query from './Query';
 import Hero from './Hero';
-// import './Home.css';
-
-const Home= () => {
-  return(
-     <Hero />
-  )
-}
-export default Home;
\ No newline at end of file
+import { readablePopulation } from './Detail';
+
+const Home = () => {
+  const [query, setQuery] = useState('');
+
+  const dispatch = useDispatch();
+  const countries = useSelector((state) => state.allCountries);
+
+  const filteredData = countries.filter((item) =>
+    item.name.toLowerCase().includes(query.toLowerCase())
+  );
+
+  useEffect(() => {
+    dispatch(getCountriesData());
+  }, [dispatch]);
+
+  const countryDetailsHandler = ({ target }) => {
+    const { id } = target;
+    dispatch(countryDetails(id));
+  };
+
+  const queryHandler = (e) => {
+    setQuery(e.target.value);
+  };
+
+  return (
+    <>
+    <Hero />
+      <Query query={query} queryChangeHandler={queryHandler} />
+        <ul className="wrap">
+          {filteredData.map((country) => (
+            <li key={country.name} className="country-card p-2">
+              <Link to="/detail" className="name">
+                <h5
+                  className="name py-2"
+                  id={country.name}
+                  onClick={countryDetailsHandler}
+                >
```

Comment:
- [ ] This function never gets the id parameter as an argument when it is called. To solve that you should use a callback function like ` onClick={() => countryDetailsHandler(country)}`

Reply:
Hello Eric! Thank you for your review! 
I believe the `countryDetailsHandler` is clarified in lines 24-27 of the Home.js file.
 `const countryDetailsHandler = ({ target }) => {
    const { id } = target;
    dispatch(countryDetails(id));
  };`